client/chat/channel-header.tsx

```diff
@@ -0,0 +1,252 @@
+    const elements = []
+    let lastIndex = 0
+
+    for (const match of sortedMatches) {
+      // This probably can't happen at this moment, but to ensure we don't get tripped by it in the
+      // future, if this happens we skip the match entirely as it means it overlaps with a previous
+      // match.
+      if (match.index < lastIndex) {
+        continue
+      }
+
+      // Insert preceding text, if any
+      if (match.index > lastIndex) {
+        elements.push(joinedChannelInfo.topic.substring(lastIndex, match.index))
+      }
+
+      if (match.type === 'link') {
+        elements.push(
+          <ExternalLink key={match.index} href={match.text}>
+            {match.text}
+          </ExternalLink>,
+        )
+      } else {
+        assertUnreachable(match.type)
+      }
+
+      lastIndex = match.index + match.text.length
+    }
+
+    // Insert remaining text, if any
+    if (joinedChannelInfo.topic.length > lastIndex) {
+      elements.push(joinedChannelInfo.topic.substring(lastIndex))
+    }
+
+    return elements
+  }, [joinedChannelInfo.topic])
+
+  const onLeaveChannelClick = useStableCallback(() => {
+    onLeaveChannel(basicChannelInfo.id)
+  })
+
+  const actions: React.ReactNode[] = []
+  if (basicChannelInfo.id !== 1 || CAN_LEAVE_SHIELDBATTERY_CHANNEL) {
+    actions.push(
+      <DestructiveMenuItem
+        key='leave-channel'
+        text={t('chat.channelHeader.actionItems.leaveChannel', 'Leave channel')}
+        onClick={onLeaveChannelClick}
+      />,
+    )
+  }
+
+  return (
+    <ChannelHeaderRoot $hasRightPadding={actions.length === 0}>
+      {!basicChannelInfo ? (
+        <LoadingDotsArea />
+      ) : (
+        <>
+          {actions.length > 0 ? (
+            <Popover
+              open={overflowMenuOpen}
+              onDismiss={closeOverflowMenu}
+              anchorX={anchorX ?? 0}
+              anchorY={anchorY ?? 0}
+              originX={'right'}
+              originY={'top'}>
+              <MenuList dense={true}>{actions}</MenuList>
+            </Popover>
+          ) : null}
+
+          <BadgeAndTextContainer>
+            <StyledChannelBadge
+              basicChannelInfo={basicChannelInfo}
+              detailedChannelInfo={detailedChannelInfo}
+            />
+            <NameAndTopicContainer>
+              <ChannelName>{basicChannelInfo.name}</ChannelName>
+              {parsedChannelTopic ? (
+                <StyledTooltip
+                  text={parsedChannelTopic}
+                  position='bottom'
+                  disabled={!isChannelTopicOverflowing}
+                  ContentComponent={StyledTooltipContent}>
+                  <ChannelTopic ref={channelTopicRef}>{parsedChannelTopic}</ChannelTopic>
+                </StyledTooltip>
+              ) : null}
+            </NameAndTopicContainer>
+          </BadgeAndTextContainer>
+
+          <ActionsArea>
+            <UserCountContainer>
+              <UserCountIcon icon='groups' size={20} />
```

Comment:
You already specified the icon in the style declaration above, so really you're just using the style declaration to declare a color now, so why is it named `UserCountIcon`? Why does it exist at all? You can just use `MaterialIcon` here, `UserCountContainer` can provide a color for it and the text (which you can also just use `Caption` for)